sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/QueryLookupOperatorConversion.java

```diff
@@ -81,12 +83,18 @@ public DruidExpression toDruidExpression(
           final DruidExpression arg = inputExpressions.get(0);
           final Expr lookupNameExpr = plannerContext.parseExpression(inputExpressions.get(1).getExpression());
           final String replaceMissingValueWith = getReplaceMissingValueWith(inputExpressions, plannerContext);
+          String lookupName = (String) lookupNameExpr.getLiteralValue();
+
+          // Put the lookup names in the query context to facilitate selective loading of lookups.
+          plannerContext.queryContextMap().putIfAbsent(PlannerContext.CTX_LOOKUPS_TO_LOAD, new HashSet<>());
+          Set<String> lookupsToLoad = (Set<String>) plannerContext.queryContextMap().get(PlannerContext.CTX_LOOKUPS_TO_LOAD);
+          lookupsToLoad.add(lookupName);
```

Comment:
```suggestion
          plannerContext.queryContextMap().computeIfAbsent(PlannerContext.CTX_LOOKUPS_TO_LOAD, key -> new HashSet<>()).add(lookupName);
```

Reply:
I'm not sure I follow. There can be multiple lookup names which have to be added to the context map. So we have to put value in the context map even if `PlannerContext.CTX_LOOKUPS_TO_LOAD` key is already present.

Please let me know if I'm misunderstanding something. Thanks!

Reply:
Yes, the suggested code will do just that. If the key is already present, it will add to the existing set. If the key is absent, it will create a new set and add the lookup to it.

Reply:
Have made the change.